File: abs_sum.py
#!/usr/bin/env python3

# Function to calculate minimum
# possible sum of all (arr[i] - b + i)
def MinSum(arr, N):

# Modify the array
	for i in range(N):
		arr[i] -= (i+1)

	# sort the array
	arr.sort()

	# calculate median
	b = arr[N//2]

	# Stores the required answer
	ans = 0
	for i in range(N):

	# Update answer
		ans += abs(arr[i]-b)

		# Return the answer
	return ans

# Driver code
arr = [1, 5, 2, -2]
N = len(arr)
print(MinSum(arr, N))

# This code is contributed by Parth Manchanda

File: abs_sum.rb
#!/usr/bin/env ruby
# frozen_string_literal: true

# Function to calculate minimum
# possible sum of all (arr[i] - b + i)
# @param arr {Array<Integer>} Array of numbers to find sum of
# @param arr {Integer} Range of array to find sum of
# @return {Integer} Minimum possible absolute sum in range
def min_sum(arr, range)
  # Modify the array
  range.times do |i|
    arr[i] -= (i + 1)
  end

  # sort the array
  arr.sort!

  # calculate median
  b = arr[range / 2]

  # Stores the required answer
  ans = 0
  range.times do |i|
    # Update answer
    ans += (arr[i] - b).abs
  end

  # Return the answer
  ans
end

# Driver code
arr = [1, 5, 2, -2]
range = arr.length
puts min_sum(arr, range)
